src/compile/data/source.ts

```diff
@@ -38,16 +37,16 @@ export namespace source {
         // For backward compatibility for former `data.formatType` property
         const formatType: DataFormat = dataFormat.type || data['formatType'];
         sourceData.format =
-          extend(
-            {type: formatType ? formatType : defaultExtension},
-            dataFormat.property ? {property: dataFormat.property} : {},
-            // Feature and mesh are two mutually exclusive properties
-            dataFormat.feature ?
-              {feature : dataFormat.feature} :
-            dataFormat.mesh ?
-              {mesh : dataFormat.mesh} :
-              {}
-          );
+        extend(
```

Comment:
the new tabbing is weird. 

How about just using the ... syntax.

const {property, feature, mesh} = dataFormat;

sourceData.format = {
  type: formatType ? formatType : defaultExtension,
  ...(property ? {property} : {}),
  ...(feature ? {feature} : {}),
  ...(mesh ? {mesh} : {}),
}